### DISCUSSION TITLE
Hit highlighting disappears with custom result source

### DISCUSSION THREAD

**Discussion by ferrarirosso** [microsoft-search/pnp-modern-search#3101] [General]
We have the following behaviour since a couple of weeks on new result sources and can reproduce this in multiple SPO tenants :
1. Create a fresh new communication site
2. Install PnP Modern Search WP in this site
3. Upload some docs with some specific test strings
4. Create a page and insert a search box connected with a search result wp, layout list and Result Source = LocalSharepointResults, so we see the HitHighlighted Summary
5. Search for one of the specific string in the doc, the search will return the results with a correct highlighted summary

Now
6. Create a new result source in the site, e.g. by copying the LocalSharepointResults, add for example the Path of the current site with +Path:https://mytenant.sharepoint.com/sites/mytestsite in the query conditions.
7. Edit the created result source id and grab the GUID after the parameter &sourceid from the URL
8. Edit the page and replace the result source id from "LocalSharepointResults" with the GUID from 7.
9. In our cases, the Hit Highlighting Summary works for a couple of minutes and then stops working, that means that the Summary is still here but without hit highlighting.

Setting the Search Results WP in debug mode, we can also see that Hit Highlightings stops working.

The following screenshots showcase when searching for the querystring testnd444*, the first one using LocalSharpeointResults, the second one using the custom result source. Both shows the result for the same doc, from the postquery operation in the dev toolbar 

### LocalSharepointResults (hit marked correctly)
![Screenshot 2023-06-27 at 14 25 26](https://github.com/microsoft-search/pnp-modern-search/assets/5904018/f4c7180d-abff-4d7a-90c1-981b490ef41e)

Custom result source (no hit highlighting)
![Screenshot 2023-06-27 at 14 23 57](https://github.com/microsoft-search/pnp-modern-search/assets/5904018/b6d02d35-9e81-4934-8469-e2c5a815a75b)

I am aware that this may be an API issue, so my questions are :
Q1 : If we submit a ticket to MS and mention that we are using PnP Modern Search, the ticket will be rejected because PnP Modern Search is not supported directly, so how should we proceed in our case ? Setting up a SPFx WP with PnPJs is probably not supported, so an SPFx WP calling the Search API Endpoint would be the right setup for that ?
Q2 : If someone can try the above case and confirm they have (no) issues with custom result sources, that would help us

Current result sources which we created "long time ago" are not impacted and are still working fine.
Please note : we may have overlooked something on our tenants, but we did double-check this issue with a couple of people and do not find what we are doing wrong
Thanks !

**Comment by wobba** [comment]
If the there are no highlights in the API response I would repro this using classic web part/UX setup, and log a ticket.

**Comment by ferrarirosso** [comment]
Thanks, a little throwback in time when playing with classic, and it shows how good & beautiful SPO is !
=> Case is opened

**Comment by ferrarirosso** [comment]
We found a workaround & solution by our own and closed the ticket.
Fyi, the problems occurs only as soon as a +Path or Path: is in the result source, from then, the HitHighlighting gets malformed (-Path or exclusion works fine).
We have replaced Path with Docaddress, the effect is the same, so we can select any sites and hit highlighting works fine
If anybody encounters the same problem, hope it helps !

**Reply by wobba** [reply]
If you do "AND Path" or add () around it, does that change anything? Weird if it works when using the alias `DocAddress` - and seems like a parsing issue/bug then to me.

**Reply by ferrarirosso** [reply]
Thanks, further investigated the issue, it works with DocAddress because we have to add `/*` at the end (with DocAddress the result source does not return any results without `/*`)
Example with DocAddress 
`{searchTerms} DocAddress:"https://mytenant.sharepoint.com/sites/mysite/*"`
Now testing with AND or enclosing parenthesis did not change anything, but, but ;-)
`{searchTerms} +Path:"https://mytenant.sharepoint.com/sites/mysite/*"
`
This works, so the problem is not +Path or DocAddress it is just that :
- The selection with +Path always works with or without trailing `/*`
- Hit Highlighting with +Path only works when adding `/*`

**Reply by wobba** [reply]
@ferrarirosso weird indeed and sounds like a snag of sorts in the underlying API.

**Reply by ferrarirosso** [reply]
if you give me the code i will have a look at it 😉, by the way is it c# ?

**Reply by spinyriptide** [reply]
@ferrarirosso DocAddress works. Path doesn't. Thank you for this workaround. This has frustrated me for some time. :)

**Comment by adamsfromthehills** [comment]
I had the same Issue and +Path only worked sometimes. However, Site works great with highlighting, with or without /*

For example:
`{searchTerms} Site:"https://mytenant.sharepoint.com/sites/mysite/"`
Hope this can help someone who is also looking for a solution
